CODE_OF_CONDUCT.md

```diff
@@ -6,14 +6,31 @@ identity and expression, sexual orientation, disability, personal
 appearance, body size, race, ethnicity, age, religion, nationality, or
 other such characteristics.
 
-Everyone is expected to follow the [Scala Code of Conduct] when
-discussing the project on the available communication channels. If you
-are being harassed, please contact us immediately so that we can
-support you.
+The ZIO project follows the [Scala Code of Conduct](https://www.scala-lang.org/conduct/), with
+adjustments with clarifications with regard to moderation that are detailed below.
+All participants, contributors and members are expected to follow the Scala Code of Conduct
+when discussing the project on the available communication channels. If you are being harassed, 
+please contact us immediately so that we can support you.
 
- ## Moderation
+## Moderation and Steering Committee
+ 
+The ZIO project's group of moderators is the Steering Committee team. The members currently are:
+* Itamar Ravid (@iravid)
+* John De Goes (@jdegoes)
+* Kai (@neko-kai)
+* Paul Shirshov (@pshirshov)
+* Salar Rahmanian (@softinio)
 
-For any questions, concerns, or moderation requests please contact a
-member of the project.
+The ZIO project requires that drastic moderation actions detailed in the code of 
+conduct - for example, removing a user from the Gitter channel - be agreed upon
+a group of over 2/3rds of the steering committee.
 
-[Scala Code of Conduct]: https://www.scala-lang.org/conduct/
+For any questions, concerns, or moderation requests please contact any member of
+the project, or the people listed above.
+
+## BDFL
+
+In addition to the above, the ZIO project's BDFL (benevolent dictator for life) is
```

Comment:
Word it without use of the bdfl title/wording? 

Other idea is for the purpose of coc we all take turns as chair of the steering committee with this responsibility when steering cannot reach a decision with a clear majority which should be very rare and hopefully never.

Reply:
Word it without use of the bdfl title/wording? 

Other idea is for the purpose of coc we all take turns as chair of the steering committee with this responsibility when steering cannot reach a decision with a clear majority which should be very rare and hopefully never.

Reply:
Just word it more neutral. John can be the tie breaker but the phrasing is just not good... "BDFL" sounds non-democratic and anti-communal. What we want is for John to support the steering committee, not be a dictator.

Reply:
Maybe Martin Odersky's role in the [Scala Improvement Process](https://docs.scala-lang.org/sips/sip-submission.html) and migration to Scala 3 is a good analogy, where he has a veto on decisions made by the committee? Hard to argue it is so terrible if it is used for governance of Scala itself.

Reply:
"BDFL" is a safe-guard mechanism that would hopefully never be exercised, but it's important to state the expectations upfront.

If the steering committee wants to do one thing, and I want to do another, and we are unable to resolve our differences of opinion, what happens? We have to do something.

As projects become successful, they are subject to hostile takeovers. I saw that with Lars taking over the Scalaz project from Tony. I would not invest in this project like I have (untold hours and significant financial expense) if that were a remote possibility.

Having a BDFL will always rub people the wrong way, but it's a protection against things that matter to me very deeply—things that my founding and participation in this project have always been contingent on.

If I ever abuse BDFL for any reason, then it will be manifestly obvious, and I'd hope everyone I count as a friend would call me on that. I also hope that my long track record of keeping cool in any situation and never lashing out at anyone, no matter what they've done to me personally, will give people the confidence and trust they need to move forward.

Reply:
Odersky himself is mentioned on the [Wikipedia page for BDFL](https://en.m.wikipedia.org/wiki/Benevolent_dictator_for_life) along with a lot of open source figures, so I don’t think it’s unreasonable John being BDFL for ZIO.

Reply:
Agree with @jdegoes lets with BDFL I say

Reply:
I agree that the title "BDFL" has an anti-communal scent to it. But, I also trust John's ability to navigate conflict responsibly and to put ego aside. I also trust his ability to take criticism should we feel he abuses this position.

I had originally suggested this clause both for the above and because I am wary of hostile takeovers. Finally - I don't think we will need to exercise this clause, having 7-8 very able personas on board on the committee.

Reply:
@iravid After reading the wikipedia entry it seems like a good fit, but... it did sound like asking for drama until I learned that is an already estabilished term. Perhaps a quote from wikiepedia explaining the term or rephrasing to something more neutral, like `founding Father of ZIO` would serve us better in getting the "right message" across?

Reply:
@iravid 
Eg.
```
## Founding Father

In addition to the above the title Founding Father of ZIO project belongs to John De Goes (@jdegoes) indefinitely, owing to him originating the project and continued investments in it. While John adheres to the same code of conduct as everyone else, he is entitled to override moderation decisions made by the steering comittee.
```